src/modules/launcher/Wox.Core/Plugin/QueryBuilder.cs

```diff
@@ -11,7 +11,7 @@ namespace Wox.Core.Plugin
 {
     public static class QueryBuilder
     {
-        public static Query Build(string text, Dictionary<string, PluginPair> nonGlobalPlugins)
+        public static Dictionary<PluginPair, Query> Build(ref string text, Dictionary<string, PluginPair> nonGlobalPlugins)
```

Comment:
I have added the ref for the query string as it removes unnecessary white spaces. I can do this in the function which is calling this but instead of doing it in every invocation, I thought it would be easier to do it in the function. The return value is the set of the plugins and their respective queries as the name of the function suggests, it builds queries so it doesn't make much sense to return the string along with those values. This string is being used for checking the cancellation token so it would prevent querying again when the queries differ only by leading/trailing or extra white spaces in between. Another thing that can be done is to do whatever is being done in this function everytime we check the cancellation token which seems like an overkill. Please let me know your thoughts because IMO it is concise this way without having to return too many things from the build function. It I were to return this value as well, it would be a tuple of a string and a dictionary which would complicate things. Also as said before since we are only removing extra white spaces I don't think it would be an issue.

Reply:
I agree that passing the cancellation token would be overkill since the function is not time-bound. Though I find mixing return and ref confusing. It's just a nit so keep as it is, if you feel otherwise.

Reply:
I'll add a function which is called prior to remove the whitespace.